### DISCUSSION TITLE
Bug? pocketbase does not see new JS hooks if pb_hooks is deleted and re-created

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#4384] [Q&A]
Not sure if this is a bug exactly, but I discovered this because I'm publishing a `pockethost` npm package that looks like this:

```
# node_modules/pockethost
package.json
dist/
   index.mjs
   app/
      pb_hooks/
         hook1.pb.js
         hook2.pb.js
```

Now when I do `npm i -g pockethost`, I think it must delete any exiting `node_modules/pockethost` before installing the latest package. That means the pb_hooks inside gets deleted.

If `pocketbase --hooksDir=/path/to/node_modules/pockethost/app/pb_hooks` was running, it seems to get the delete event, but does not pick up the new files. I assume it's watching the old inode for changes and doesn't see the new one?

The effect is that all my custom routes suddenly start returning 404. Restarting pocketbase cures it.

**Comment by ganigeorgiev** [comment]
There many reasons why this may not work.

First note that the watcher works only when the content of `pb_hooks` is modified.

Additionally, if on process bootstrap there is no hooks directory the file watcher is not initialized (it is done mostly for performance reason).

Another thing that may be also the reason why the file watcher doesn't work depending on your operating system could be a remaining inotify descriptor because you are deleting the folder while the application is still running, meaning that technically it may not be deleted by the os until the process terminate (the descriptor is closed) and there are no other descriptors pointing to the folder/file.
